### ISSUE TITLE
Dropdown is behind as z-index doesn't apply

### ISSUE THREAD

**Issue by zippoxer** [Dogfalo/materialize#1385]
This issue doesn't apply to the current release version 0.96.1, only to the current master branch.

The problem is that a (not yet released) change is attaching the dropdown element to the activator element, therefore forcing the dropdown menu to adopt the z-index of the activator's parent causing it to be partially hidden in various cases.

Previously, I was placing dropdown elements inside the body element, they were showing in the right place and above anything as dropdown elements have z-index of 999.
Now, dropdown elements, no matter where you put them, end up being next to their activators and therefore adopting the z-index of their activator's parent.

In my case, I have a list of boxes, all with identical z-index (well below the defined 999 of a dropdown menu) and a menu button at their right edge. Since the dropdown adopts the box's z-index, the dropdown is always hidden by the next box in the list, as the next box was appended later to the page and therefore appears above the previous.
![dropdown-issue](https://cloud.githubusercontent.com/assets/859015/7712408/e91cfe9c-fe79-11e4-8ef1-227a18e2988d.png)

**Comment by ckdt** [comment]
Same issue over here

**Comment by zippoxer** [comment]
This issue also causes another problem, I just discovered. The dropdown menu is cut or hidden if it's parent element has `overflow: hidden`:

![dropdown-issue2](https://cloud.githubusercontent.com/assets/859015/7785412/75fb951a-0197-11e5-84e9-ec7ac6fdf5cf.png)

**Comment by jcapogna** [comment]
I think was fixed in 0.97.0. I just upgraded from 0.96.1 to 0.97.0 specifically to fix this problem (my dropdown was showing underneath my submit button).

**Comment by carlosperate** [comment]
I'm using a new version than 0.97.0 (js grabbed from master) and my dropdowns are still behind the modal footer (from modal with fixed footer) with a scrollbar, as the materialize select `ul` element is attached to the parent.

**Comment by nathanjohnson320** [comment]
I had a similar issue but the dropdown was being hidden by overflow (i.e. inside a .card) so I just changed overflow of the parent element to visible and that fixed it.

**Comment by exp0nge** [comment]
Same issue for when dropdown is inside a collection.

**Comment by osxi** [comment]
@nathanjohnson320's suggestion helped me fix the issue.

In my case, I had a select inside of a `div class="card"` and setting the card to something other than `overflow: none;` caused the overflow contents to show up as you would expect.

**Comment by serkandurusoy** [comment]
I'm using dropdown within a collection and facing the same problem. 

```css
.collection {
  overflow: visible;
}
```

solved it in my case, but I don't know if this messes with something else.

**Comment by osxi** [comment]
@serkandurusoy, that's a good point. To prevent that, you could just use a class like `contacts-collection` and use something along the lines of:
```css
.contacts-collection {
  @extend .collection;
  overflow: visible;
}
```
This way, other collections won't be affected.

**Comment by serkandurusoy** [comment]
@osxi thanks for the suggestion.

**Comment by whatda** [comment]
@nathanjohnson320 `overflow-y: visible` on the `.modal` class worked well for me, too!

**Comment by winiram** [comment]
![screen shot 2016-04-28 at 5 32 05 pm](https://cloud.githubusercontent.com/assets/7262768/14905616/3270439a-0d67-11e6-873a-716cb8d083de.png)

This is happening. I am using 0.97.0. Any suggestions?

**Comment by lordplagus02** [comment]
Any news on this?

**Comment by weinde** [comment]
I'm having the same issue with my project... any news on this?

**Comment by weinde** [comment]
Has the issue been resolved?

**Comment by weinde** [comment]
So has anyone found a slolution to this? I'm having the same issues... 
![screenshot from 2016-10-28 08_46_58](https://cloud.githubusercontent.com/assets/23027747/19797902/2225cd22-9cee-11e6-9bc6-f19f9765f6eb.jpg)
![screenshot from 2016-10-24 11_18_29](https://cloud.githubusercontent.com/assets/23027747/19797903/2226081e-9cee-11e6-9e83-77c4ee33735e.jpg)

**Comment by IaraNascimento** [comment]
I had the same problem: when i open the options/dropdown from a select inside a card the options appear behind the other cards in the page. I tried change the z-index, and it looked like was going to work... but it didn't. I put one card with a bigger z-index and all selects inside it gets in front of the others cards, and them i find myself in same problem, because the others cards also had selects and now they were behind the one that i just had increased the z-index.

![1](https://cloud.githubusercontent.com/assets/6676988/20367627/72505490-ac37-11e6-8052-a9f2d58e6ab4.png)

![2](https://cloud.githubusercontent.com/assets/6676988/20367737/f1d76bc2-ac37-11e6-965b-3a5121f29fd3.png)

So i tried make the z-index change more dynamically using hover. Like this:

.card {
	z-index: 0;
}
.card:hover {
	z-index: 1 !important;
}

Now it is better... but is still get weird when the mouse goes out of the card with the select open. So i think i'll do a js with mouse-in ando mouse-out. But for now i'll use this second solution.

**Comment by alan-xie** [comment]
The `:hover` response above doesn't work for my particular implementation (I have multiple selects in a single modal). 

However, I've added the below snippet to my `materializeHandler()` global helper function, which I use to run all of the jQuery inits. All of my select elements are currently wrapped in a div with the `select-wrapper` class, which is what I'm referencing here. This dynamically sets the CSS of each of the parent wrappers on initialization such that the earlier selects in the page will always have a higher z-index than the later ones.

```
$(".select-wrapper").each(function(i) {
    $(this).css({"z-index" : 500 - i})
})
```

**Comment by nivinjoseph** [comment]
This might be a stacking context issue as mentioned in this article https://philipwalton.com/articles/what-no-one-told-you-about-z-index/

I had a similar problem with card in main area overlapping dropdown-content in the top navbar. Setting the z-index of the navbar, which in my case was the parent of the dropdown-content, to 1 solved the issue.

**Comment by geekDago** [comment]
I'm having this problem again: _overflow-y: visible_ on the .modal seemed to be the solution, but if one reduces the windows size or use it on mobile, the modal content "flows" over the screen outside the modal

**Comment by acburst** [comment]
Changed z-index of dropdown in 42aa7476.

**Comment by joaorsfonseca** [comment]
This is not fixed on last version 0.100.2..

![image](https://user-images.githubusercontent.com/33959685/35453193-39eb7404-02c1-11e8-901d-b4af59adf335.png)

**Comment by smadapathi** [comment]
We still face the same issue mentioned by @joaorsfonseca

**Comment by acburst** [comment]
We are not planning anymore releases to the v0.100.x version. I meant its fixed on our v1.0.0 alpha branch. Sorry for the confusion

**Comment by smadapathi** [comment]
any workaround for this bug?

**Comment by MohammedAl-Rowad** [comment]
~~ JQuery Solution ~~
~~ by changing the z-index to the elements that the drop down shows behind it, to -1 when an element trigger the drop down menu ~~
~~ and when it's closed just return the z-index to 1 ~~
I'm sorry if I shouldn't write this at this time(scene the question was posted 3 years ago, but it looks like some people still facing this problem like me ) .
I manage to solve this using JQuery  
(the drop down in the Navbar was behind the cards)

Image 

![screenshot 26](https://user-images.githubusercontent.com/38977667/43044171-2671e4d4-8daa-11e8-980b-e99542d7ed0e.png)


the solution:
`    // init the navbar
    $(".dropdown-trigger").dropdown({
        onCloseStart : () => {
            $('.card').css('z-index', 1)
        }
   });`

`
    // the <li>  that opens the dropdown list in the navbar
    $("#fixOverFlowIssue").click(()=>{
        $('.card').css('z-index',-1)
    })
`

read more : [here](https://materializecss.com/dropdown.html)

**Comment by khullah** [comment]
Nothing above worked for me (due to project particularities). My solution was to overcome the stacking context, increasing the parent's (and in some cases grand-parent's) z-index value.

**Comment by timothymarois** [comment]
Having the same issue. Not seeing a solution. Why is this even a problem, people have already done this and already solved these issues with matured versions of js plugins like Chosen, Select2 etc.

**Comment by simkessy** [comment]
@timothymarois agreed, not even sure if this is really maintained anymore.

**Comment by amoliski** [comment]
They claim to have fixed it in 1.0.0, but I'm still having this problem:

https://codepen.io/anon/pen/oaJOEj

**Comment by jnorton2** [comment]
This was our solution:
`.select-dropdown` {
    top: 0 !important;
}`

**Comment by ronpetit** [comment]
> They claim to have fixed it in 1.0.0, but I'm still having this problem:
> 
> https://codepen.io/anon/pen/oaJOEj
> 
> Edit: Cleared it up. Have to say this is not intuitive at all
> 
> ```
> var elem = document.getElementById('select_dropdown');
> M.FormSelect.init(elem, {dropdownOptions:{container:document.body}});
> ```

Really it is not, this should be pointed out in the documentation
